spec/lib/efile/nj/nj1040_spec.rb

```diff
@@ -0,0 +1,144 @@
+require 'rails_helper'
+
+describe Efile::Nj::Nj1040 do
+  let(:intake) { create(:state_file_nj_intake) }
+  let(:instance) do
+    described_class.new(
+      year: MultiTenantService.statefile.current_tax_year,
+      intake: intake
+    )
+  end
+
+  describe '#calculate_line_6' do
+    context 'when filer has no spouse' do
+      before do
+        intake.direct_file_data.filing_status = 1 # single
+        instance.calculate
+      end
+      it { expect(instance.lines[:NJ1040_LINE_6].value).to eq(1000) }
+    end
+
+    context 'when filer has a spouse' do
+      let(:intake) { create(:state_file_nj_intake, :married) }
+      before do
+        instance.calculate
+      end
+      it { expect(instance.lines[:NJ1040_LINE_6].value).to eq(2000) }
+    end
+  end
+
+  describe '#calculate_line_7' do
```

Comment:
I think instead of trying to separate each of these calculations with the `describe '#calculate_line_7'` I might try to simplify it by just having the larger context blocks of which filing status they have (or other circumstances that come up) and then put the expectations of what the line values should be in there. Additionally, there doesn't necessarily need to be a separate it statement for each one, they could be grouped together and hopefully that will help with making the test file more readable and easier to manage as y'all continue to add more items

Reply:
Thanks! Appreciate the guidance on grouping multiple `expect` blocks together. I was trying to follow https://www.betterspecs.org/#single but agree that it makes the test way too long 😭